### PR TITLE
python3Packages.defusedxml: 0.6.0 -> 0.7.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.7.0

Add `meta`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115251` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

File: pkgs/development/python-modules/defusedxml/default.nix [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+  version = "0.7.0";
+  disabled = pythonOlder "3.6";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "f684034d135af4c6cbb949b8a4d2ed61634515257a67299e5f940fbaa34377f5";
+    sha256 = "183fz8xwclhkirwpvpldyypn47r8lgzfz2mk9jgyg7b37jg5vcc6";
+  };
+
+  pythonImportsCheck = [ "defusedxml" ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/defusedxml/default.nix:17]: Enable tests by passing `tests.py` to the pytestCheckHook.

python3Packages.defusedxml

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115251` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>ihaskell</li>
<li>python38Packages.optuna</li>
<li>python38Packages.pymc3</li>
<li>python38Packages.skein</li>
<li>python38Packages.skorch</li>
<li>python38Packages.textacy</li>
<li>python39Packages.skein</li>
</ul>
</details>
<details>

  <summary>23 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>ansible-lint: log was empty</li>
<li>(python38Packages.ansible-lint): log was empty</li>
<li>apache-airflow: log was empty</li>
<li>etesync-dav: log was empty</li>
<li>mailman-web: log was empty</li>
<li>(python38Packages.mailman-web): log was empty</li>
<li>python38Packages.flask-appbuilder: log was empty</li>
<li>python38Packages.gradient: log was empty</li>
<li>python38Packages.gradient-utils: log was empty</li>
<li>python38Packages.hyperkitty: log was empty</li>
<li>python38Packages.jupyterlab-git: log was empty</li>
<li>python38Packages.nbdime: log was empty</li>
<li>python38Packages.pywick: log was empty</li>
<li>python38Packages.shap: log was empty</li>
<li>spyder: log was empty</li>
<li>(python38Packages.spyder): log was empty</li>
<li>python38Packages.sunpy: log was empty</li>
<li>python39Packages.flask-appbuilder: log was empty</li>
<li>python39Packages.hyperkitty: log was empty</li>
<li>python39Packages.jupyterlab-git: log was empty</li>
<li>python39Packages.mailman-web: log was empty</li>
<li>python39Packages.nbdime: log was empty</li>
<li>s3ql: log was empty</li>
  </ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>python38Packages.datashader: log was empty</li>
  </ul>
</details>
<details>
<summary>
243 packages built:
</summary>
<ul>
<li>aiodnsbrute</li>
<li>backblaze-b2</li>
<li>calibre</li>
<li>check_systemd</li>
<li>checkov</li>
<li>conan</li>
<li>cq-editor</li>
<li>deja-dup</li>
<li>devpi-client</li>
<li>devpi-server</li>
<li>diffoscope</li>
<li>duplicity</li>
<li>duply</li>
<li>evdevremapkeys</li>
<li>evillimiter</li>
<li>fdroidserver</li>
<li>ffmpeg-normalize</li>
<li>gdown (python38Packages.gdown)</li>
<li>gnome-keysign</li>
<li>home-assistant</li>
<li>img2pdf (python38Packages.img2pdf)</li>
<li>jellyfin-mpv-shim</li>
<li>jira-cli</li>
<li>jupyter</li>
<li>kaggle (python38Packages.kaggle)</li>
<li>keepkey_agent (python38Packages.keepkey_agent)</li>
<li>ldeep</li>
<li>ledger_agent (python38Packages.ledger_agent)</li>
<li>lexicon</li>
<li>lieer</li>
<li>luigi</li>
<li>magic-wormhole (python38Packages.magic-wormhole)</li>
<li>manim</li>
<li>mathlibtools (python38Packages.mathlibtools)</li>
<li>matrix-synapse</li>
<li>mirage-im</li>
<li>mnemosyne</li>
<li>noto-fonts-emoji</li>
<li>nrfutil</li>
<li>ocrfeeder</li>
<li>ocrmypdf</li>
<li>papis (python38Packages.papis)</li>
<li>passExtensions.pass-import</li>
<li>pdfarranger</li>
<li>piston-cli</li>
<li>plex-mpv-shim</li>
<li>python38Packages.Nikola</li>
<li>python38Packages.Pweave</li>
<li>python38Packages.afdko</li>
<li>python38Packages.ansible-kernel</li>
<li>python38Packages.ansible-runner</li>
<li>python38Packages.async-upnp-client</li>
<li>python38Packages.b2sdk</li>
<li>python38Packages.batchspawner</li>
<li>python38Packages.bugwarrior</li>
<li>python38Packages.cairosvg</li>
<li>python38Packages.connect-box</li>
<li>python38Packages.cufflinks</li>
<li>python38Packages.datasets</li>
<li>python38Packages.defusedxml</li>
<li>python38Packages.dipy</li>
<li>python38Packages.django-allauth</li>
<li>python38Packages.django-mailman3</li>
<li>python38Packages.dnachisel</li>
<li>python38Packages.dockerspawner</li>
<li>python38Packages.exchangelib</li>
<li>python38Packages.flask-openid</li>
<li>python38Packages.genome-collector</li>
<li>python38Packages.graphtage</li>
<li>python38Packages.gtts</li>
<li>python38Packages.holoviews</li>
<li>python38Packages.hvplot</li>
<li>python38Packages.hyperopt</li>
<li>python38Packages.ignite</li>
<li>python38Packages.intake</li>
<li>python38Packages.internetarchive</li>
<li>python38Packages.ipydatawidgets</li>
<li>python38Packages.ipympl</li>
<li>python38Packages.ipyvue</li>
<li>python38Packages.ipyvuetify</li>
<li>python38Packages.ipywidgets</li>
<li>python38Packages.jira</li>
<li>python38Packages.jupyter</li>
<li>python38Packages.jupyter-sphinx</li>
<li>python38Packages.jupyterhub</li>
<li>python38Packages.jupyterhub-ldapauthenticator</li>
<li>python38Packages.jupyterhub-systemdspawner</li>
<li>python38Packages.jupyterhub-tmpauthenticator</li>
<li>python38Packages.jupyterlab</li>
<li>python38Packages.jupyterlab_launcher</li>
<li>python38Packages.jupyterlab_server</li>
<li>python38Packages.kmapper</li>
<li>python38Packages.labelbox</li>
<li>python38Packages.libagent</li>
<li>python38Packages.lightparam</li>
<li>python38Packages.livelossplot</li>
<li>python38Packages.lsassy</li>
<li>python38Packages.mesa</li>
<li>python38Packages.modeled</li>
<li>python38Packages.moretools</li>
<li>python38Packages.moviepy</li>
<li>python38Packages.msldap</li>
<li>python38Packages.myfitnesspal</li>
<li>python38Packages.nagiosplugin</li>
<li>python38Packages.nbclient</li>
<li>python38Packages.nbconflux</li>
<li>python38Packages.nbconvert</li>
<li>python38Packages.nbsmoke</li>
<li>python38Packages.nbsphinx</li>
<li>python38Packages.ndjson</li>
<li>python38Packages.nix-kernel</li>
<li>python38Packages.nltk</li>
<li>python38Packages.notebook</li>
<li>python38Packages.notedown</li>
<li>python38Packages.nototools</li>
<li>python38Packages.oauthenticator</li>
<li>python38Packages.odfpy</li>
<li>python38Packages.panel</li>
<li>python38Packages.papermill</li>
<li>python38Packages.parfive</li>
<li>python38Packages.perfplot</li>
<li>python38Packages.phik</li>
<li>python38Packages.pikepdf</li>
<li>python38Packages.plexapi</li>
<li>python38Packages.plugwise</li>
<li>python38Packages.postorius</li>
<li>python38Packages.privacyidea</li>
<li>python38Packages.proglog</li>
<li>python38Packages.pyexcel-ods</li>
<li>python38Packages.pygal</li>
<li>python38Packages.pyppeteer</li>
<li>python38Packages.pypykatz</li>
<li>python38Packages.pysaml2</li>
<li>python38Packages.python-daemon</li>
<li>python38Packages.python-didl-lite</li>
<li>python38Packages.python-miio</li>
<li>python38Packages.python-mpv-jsonipc</li>
<li>python38Packages.python3-openid</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.rbtools</li>
<li>python38Packages.rich</li>
<li>python38Packages.rising</li>
<li>python38Packages.robotframework-tools</li>
<li>python38Packages.rxv</li>
<li>python38Packages.sacremoses</li>
<li>python38Packages.salmon-mail</li>
<li>python38Packages.spacy</li>
<li>python38Packages.spyder_3</li>
<li>python38Packages.stytra</li>
<li>python38Packages.surepy</li>
<li>python38Packages.tablib</li>
<li>python38Packages.thinc</li>
<li>python38Packages.toggl-cli</li>
<li>python38Packages.tqdm</li>
<li>python38Packages.transformers</li>
<li>trezor_agent (python38Packages.trezor_agent)</li>
<li>python38Packages.tubeup</li>
<li>python38Packages.vega</li>
<li>python38Packages.weasyprint</li>
<li>python38Packages.widgetsnbextension</li>
<li>xml2rfc (python38Packages.xml2rfc)</li>
<li>python38Packages.zeep</li>
<li>python38Packages.zetup</li>
<li>python39Packages.Nikola</li>
<li>python39Packages.Pweave</li>
<li>python39Packages.async-upnp-client</li>
<li>python39Packages.batchspawner</li>
<li>python39Packages.cairosvg</li>
<li>python39Packages.connect-box</li>
<li>python39Packages.cufflinks</li>
<li>python39Packages.defusedxml</li>
<li>python39Packages.django-allauth</li>
<li>python39Packages.django-mailman3</li>
<li>python39Packages.dockerspawner</li>
<li>python39Packages.exchangelib</li>
<li>python39Packages.flask-openid</li>
<li>python39Packages.img2pdf</li>
<li>python39Packages.ipydatawidgets</li>
<li>python39Packages.ipympl</li>
<li>python39Packages.ipyvue</li>
<li>python39Packages.ipyvuetify</li>
<li>python39Packages.ipywidgets</li>
<li>python39Packages.jira</li>
<li>python39Packages.jupyter</li>
<li>python39Packages.jupyter-sphinx</li>
<li>python39Packages.jupyterhub</li>
<li>python39Packages.jupyterhub-ldapauthenticator</li>
<li>python39Packages.jupyterhub-systemdspawner</li>
<li>python39Packages.jupyterhub-tmpauthenticator</li>
<li>python39Packages.jupyterlab</li>
<li>python39Packages.jupyterlab_launcher</li>
<li>python39Packages.jupyterlab_server</li>
<li>python39Packages.kmapper</li>
<li>python39Packages.lightparam</li>
<li>python39Packages.livelossplot</li>
<li>python39Packages.modeled</li>
<li>python39Packages.moretools</li>
<li>python39Packages.nbclient</li>
<li>python39Packages.nbconflux</li>
<li>python39Packages.nbconvert</li>
<li>python39Packages.nbsphinx</li>
<li>python39Packages.nix-kernel</li>
<li>python39Packages.notebook</li>
<li>python39Packages.notedown</li>
<li>python39Packages.oauthenticator</li>
<li>python39Packages.odfpy</li>
<li>python39Packages.pikepdf</li>
<li>python39Packages.plugwise</li>
<li>python39Packages.postorius</li>
<li>python39Packages.privacyidea</li>
<li>python39Packages.pyexcel-ods</li>
<li>python39Packages.pygal</li>
<li>python39Packages.pysaml2</li>
<li>python39Packages.python-didl-lite</li>
<li>python39Packages.python3-openid</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.robotframework-tools</li>
<li>python39Packages.rxv</li>
<li>python39Packages.tablib</li>
<li>python39Packages.vega</li>
<li>python39Packages.widgetsnbextension</li>
<li>python39Packages.zeep</li>
<li>python39Packages.zetup</li>
<li>r128gain</li>
<li>radicale (radicale3)</li>
<li>sacad</li>
<li>sourcehut.metasrht</li>
<li>tartube</li>
<li>theharvester</li>
<li>tinyprog</li>
<li>twitter-color-emoji</li>
<li>urlwatch</li>
<li>xandikos</li>
</ul>
</details>

File: pkgs/development/python-modules/defusedxml/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/defusedxml/default.nix:17]: Not required https://github.com/tiran/defusedxml/blob/master/tox.ini#L42

Comment (mweinelt) [reply pkgs/development/python-modules/defusedxml/default.nix:17]:
```diff
diff --git a/pkgs/development/python-modules/defusedxml/default.nix b/pkgs/development/python-modules/defusedxml/default.nix
index 3abdd0fb792..ef1ba91d45c 100644
--- a/pkgs/development/python-modules/defusedxml/default.nix
+++ b/pkgs/development/python-modules/defusedxml/default.nix
@@ -2,6 +2,7 @@
 , buildPythonPackage
 , fetchPypi
 , pythonOlder
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
@@ -14,6 +15,8 @@ buildPythonPackage rec {
     sha256 = "183fz8xwclhkirwpvpldyypn47r8lgzfz2mk9jgyg7b37jg5vcc6";
   };
 
+  checkInputs = [ pytestCheckHook ];
+
   pythonImportsCheck = [ "defusedxml" ];
 
   meta = with lib; {
```

```
=========================== short test summary info ============================
FAILED tests.py::BaseTests::test_allow_expansion - AttributeError: 'BaseTests...
FAILED tests.py::BaseTests::test_dtd_forbidden - AttributeError: 'BaseTests' ...
FAILED tests.py::BaseTests::test_dtd_with_external_ref - AttributeError: 'Bas...
FAILED tests.py::BaseTests::test_entities_forbidden - AttributeError: 'BaseTe...
FAILED tests.py::BaseTests::test_entity_cycle - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_external_file_ref - AttributeError: 'BaseTes...
FAILED tests.py::BaseTests::test_external_ref - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse - AttributeError: 'BaseTests' o...
FAILED tests.py::BaseTests::test_simple_parse_ns - AttributeError: 'BaseTests...
FAILED tests.py::TestDefusedElementTree::test_html_arg - AssertionError: Depr...
FAILED tests.py::TestDefusedcElementTree::test_html_arg - AssertionError: Dep...
FAILED tests.py::TestDefusedLxml::test_allow_expansion - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_dtd_forbidden - TypeError: catching cl...
FAILED tests.py::TestDefusedLxml::test_dtd_with_external_ref - TypeError: cat...
FAILED tests.py::TestDefusedLxml::test_entities_forbidden - TypeError: catchi...
FAILED tests.py::TestDefusedLxml::test_entity_cycle - TypeError: assertRaises...
FAILED tests.py::TestDefusedLxml::test_lxml_warnings - AssertionError: None i...
FAILED tests.py::TestDefusedLxml::test_restricted_element1 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_restricted_element2 - TypeError: catch...
FAILED tests.py::TestDefusedLxml::test_simple_parse - TypeError: catching cla...
FAILED tests.py::TestDefusedLxml::test_simple_parse_ns - TypeError: catching ...
FAILED tests.py::TestDefusedLxml::test_xpath_injection - AttributeError: 'Non...
================== 22 failed, 67 passed, 2 warnings in 0.91s ===================
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/defusedxml/default.nix:17]: and fails with lots of type errors.

Comment (mweinelt) [reply pkgs/development/python-modules/defusedxml/default.nix:17]: Uh, I'm a bit baffled why you merge this anyway. The tests are working upstream.

https://github.com/tiran/defusedxml/runs/2029830851

Comment (fabaff) [reply pkgs/development/python-modules/defusedxml/default.nix:17]: @SuperSandro2000, made the requested changes in a new PR https://github.com/NixOS/nixpkgs/pull/115341

**Pull request merged**